src/app/auth/interceptors/error.interceptor.ts

```diff
@@ -0,0 +1,24 @@
+import { HttpInterceptor } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { HttpRequest, HttpHandler, HttpEvent } from '@angular/common/http';
+import { Observable, throwError } from 'rxjs';
+import { catchError } from 'rxjs/operators';
+import { AuthService } from '../services/auth.service';
+
+@Injectable()
+export class ErrorInterceptor implements HttpInterceptor {
+  constructor(private authService: AuthService) {}
+
+  intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+    return next.handle(req).pipe(
+      catchError((err) => {
+        if ([401, 403].indexOf(err.status) !== -1) {
```

Comment:
403 might not mean that the user is not authenticated, but it means user is not authorized. They probably may miss some permissions for particular resources

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403.



And I'm not saying to change it, but it appears to me that [401, 403].includes(err.status) would be much more readable